src/main/java/seedu/address/storage/JsonAdaptedExpenditure.java

```diff
@@ -0,0 +1,44 @@
+package seedu.address.storage;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.entry.*;
+import seedu.address.model.tag.Tag;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Jackson-friendly version of {@link Expenditure}.
+ */
+public class JsonAdaptedExpenditure extends JsonAdaptedEntry {
+    /**
+     * Constructs a {@code JsonAdaptedEntry} with the given person details.
+     */
+    @JsonCreator
+    public JsonAdaptedExpenditure(@JsonProperty("description") String description, @JsonProperty("amount") String amount,
+                            @JsonProperty("date") String date, @JsonProperty("tagged") List<JsonAdaptedTag> tagged) {
+        super(description, amount, date, tagged);
+    }
+
+    public JsonAdaptedExpenditure(Entry source) {
+        super(source);
+    }
+
+    @Override
+    public Expenditure toModelType() throws IllegalValueException {
+        checkIsValidJsonEntry();
+        final List<Tag> personTags = new ArrayList<>();
+        for (JsonAdaptedTag tag : tagged) {
```

Comment:
LGTM for the storage saving features, and the separation of the entries into `Income` and `Expenditure` classes, I just need to fix the naming from persons, address books -> entries and Penny wise but we can do that after we merge the commits in so the rest can work on their features as well.